schema/bom-1.5.xsd

```diff
@@ -3830,37 +3924,1298 @@ limitations under the License.
+                        CycloneDX Property Taxonomy - https://github.com/CycloneDX/cyclonedx-property-taxonomy.
+                        Formal registration is OPTIONAL.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded">
+                <xs:annotation>
+                    <xs:documentation>
+                        Allows any undeclared elements as long as the elements are placed in a different namespace.
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:any>
+        </xs:sequence>
+        <xs:anyAttribute namespace="##any" processContents="lax">
+            <xs:annotation>
+                <xs:documentation>User-defined attributes may be used on this element as long as they
+                    do not have the same name as an existing attribute used by the schema.</xs:documentation>
+            </xs:annotation>
+        </xs:anyAttribute>
+    </xs:complexType>
+
+    <xs:complexType name="inputType">
+        <xs:annotation>
+            <xs:documentation>
+                Type that represents various input data types and formats.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:choice>
+                <xs:element name="resource" type="bom:resourceReferenceType" minOccurs="1" maxOccurs="1">
+                    <xs:annotation>
+                        <xs:documentation>
+                            A reference to an independent resource provided as an input to a task by the workflow runtime.
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+                <xs:element name="parameters" type="bom:parametersType" minOccurs="1" maxOccurs="1">
+                    <xs:annotation>
+                        <xs:documentation>
+                            Inputs that have the form of parameters with names and values.
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+                <xs:element name="environmentVars" minOccurs="1" maxOccurs="1">
+                    <xs:annotation>
+                        <xs:documentation>
+                            Inputs that have the form of parameters with names and values.
+                        </xs:documentation>
+                    </xs:annotation>
+                    <xs:complexType>
+                        <xs:sequence minOccurs="0" maxOccurs="unbounded">
+                            <!-- maxOccurs="unbounded" NEEDS to be set on the sequence, not the individual elements -->
+                            <xs:choice>
+                                <xs:element name="environmentVar" type="bom:propertyType" minOccurs="0" maxOccurs="1"/>
+                                <xs:element name="value" type="xs:string" minOccurs="0" maxOccurs="1"/>
```

Comment:
From JSON schema... the choice for `inputs` is:
```
  "inputType": {
            "type": "object",
            "oneOf": [
                {
                    "required": [
                        "resource"
                    ]
                },
                {
                    "required": [
                        "parameters"
                    ]
                },
                {
                    "required": [
                        "environmentVars"
                    ]
                },
                {
                    "required": [
                        "data"
                    ]
                }
            ],
```

where env. vars is:
```
                "environmentVars": {
                    "type": "array",
                    "items": {
                        "oneOf": [
                            {
                                "$ref": "#/definitions/property"
                            },
                            {
                                "type": "string"
                            }
                        ]
                    },
                    "uniqueItems": true
                },
```

where (yes) env. vars. would be captured as strings (any non-string interpretation is processor dependent).

.... perhaps this addresses similar "string" questions?

Reply:
current implementation would allow the following:
```xml
<environmentVars>
  <environmentVar><!-- no content --></environmentVar>
  <environmentVar name="foo"><!-- no value --></environmentVar>
  <environmentVar>bar<!-- no key --></environmentVar>
  <value>foo=bar; a=df
<!-- arbitrary string that looks like a collection of vars --></environmentVar>
  <value>
    foo2: bar
    a2: df
<!-- arbitrary string that looks that might look loke a collection of vars --></environmentVar>
</environmentVars>
```
- no value - makes no sense.
- no key - makes no sense;
- all vars as one blob - the encoding of these is undefined, so everybody would do what they want and we would have incompatible results, right?

----

i'd suggest writing the schema in a way for 
```xml
<environmentVars>
  <environmentVar name="foo">bar</environmentVar>
  <environmentVar name="baz"><![CDATA[lorem imsum
dolor sit amet.
]]></environmentVar>
</environmentVars>
```
`name` is mandatory arg of type "token" with minlength=1; content is of type "string". no element `value` allowed.

Reply:
see #222